Implemented getting output dir from eleventyConfig & manual override

Denperidge opened the pull request:
Hi! Small addition, but it decreases the setup work needed.

Denperidge committed:
Implemented getting output dir from eleventyConfig & manual override

NJAldwin commented:
Hi, thanks for the PR!  That does seem like a nice addition.

I will have some more time to review this in depth a bit later.  I have a couple of small questions:

1. For backwards compatibility, can we preserve the original name of `outputDir`?
2. I seem to remember having some issue with this hence being why it's not included.  Perhaps that's not an issue anymore, but would it be possible to add a test of the behaviors just to be safe?

Thank you!

Denperidge commented:
1. Makes perfect sense!
2. As in, testing that the parameter for sure overrides the eleventy.dir.output usage? Will already whip something like that up, but let me know if you meant something different

Edit: is it possible to test eleventy config behaviours in the current testing setup? because genFavicons doesn't handle the default-dir behaviour, but index.js instead

Denperidge committed:
overrideOutputDir -> outputDir

See https://github.com/NJAldwin/eleventy-plugin-gen-favicons/pull/14#issuecomment-2520525514

Denperidge committed:
WIP: eleventy-test integration

One of the scenarios outputs on _site, which it shouldn't

Denperidge committed:
eleventyConfig.dir -> eleventyConfig.directories

Good news:
- eleventy-test wasn't malfunctioning; the bug it gave was an actual bug in my code addition here
- NJAldwin was fully correct with point 2.
https://github.com/NJAldwin/eleventy-plugin-gen-favicons/pull/14#issuecomment-2520525514
eleventyConfig.dir does not keep CLI in check
However, eleventyConfig.directories does

Denperidge committed:
WIP: fix async in eleventy-test

Denperidge committed:
Finished implementing eleventy-test

Denperidge commented:
Hi hi! Excuse the many WIP commits (I can squash them together if wanted), but I've added tests for it by using eleventy-test (which I have just made, admittedly) and even figured out what your previous past issues might've been from!

If this test sounds okay for you:
- I will make sure it also works on Eleventy 2 by expanding the tests
- I can optionally add tests in there for plugin-in-Eleventy testing!
